metis/lib/models/folder.rb

```diff
@@ -97,6 +97,28 @@ def rename!(new_folder, new_folder_name)
       refresh
     end
 
+    def update_bucket!(new_bucket)
+      update(bucket: new_bucket)
+
+      # Need to recursively update all sub-folders and files
+      files.each { |file|
+        file.update_bucket!(new_bucket)
+      }
+      folders.each { |folder|
+        folder.update_bucket!(new_bucket)
+      }
+
+      refresh
+    end
+
+    def update_bucket_and_rename!(new_folder, new_folder_name, new_bucket)
+      # rename first to avoid any potential collisions with folder names
```

Comment:
Does, this logic work?  What if the rename! conflicts locally but not in the remote bucket?  

I want to rename my_bucket/a/b to other_bucket/c/d, but there happens to be a my_bucket/c/d.  Won't this fail although it does not have to?

Reply:
I think you cannot break this down into 2 discrete steps.  I think the validation only works if the bucket and renaming happen together at the same time.

Reply:
Thanks, great catch!